src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,124 @@
 package co.inventorsoft.academy.collections.model;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Function;
 
 public class Range<T> implements Set<T> {
 
+    private HashSet<T> range;
+
+    public Range(){
+        range = new HashSet<T>();
+    }
+
     public int size() {
-        return 0;
+        return range.size();
     }
 
     public boolean isEmpty() {
-        return false;
+        if(size()==1){
+            return true;
+        }
+        return range.isEmpty();
     }
 
     public boolean contains(Object o) {
-        return false;
+        return range.contains(o);
     }
 
     public Iterator<T> iterator() {
-        return null;
+        return range.iterator();
     }
 
     public Object[] toArray() {
-        return new Object[0];
+        return range.toArray();
     }
 
     public <T1> T1[] toArray(T1[] a) {
-        return null;
+        return (T1[]) range.toArray(a);
     }
 
     public boolean add(T t) {
-        return false;
+        return range.add(t);
     }
 
     public boolean remove(Object o) {
-        return false;
+        return range.remove(o);
     }
 
     public boolean containsAll(Collection<?> c) {
-        return false;
+        return range.containsAll(c);
     }
 
     public boolean addAll(Collection<? extends T> c) {
-        return false;
+        return range.addAll(c);
     }
 
     public boolean retainAll(Collection<?> c) {
-        return false;
+        return range.retainAll(c);
     }
 
     public boolean removeAll(Collection<?> c) {
-        return false;
+        return range.removeAll(c);
     }
 
     public void clear() {
+        range.clear();
+    }
+
+    public static Range<Byte> of( byte element1, byte element2){
+        Range<Byte> rByte = new Range<>();
+        for(byte i=element1;i<=element2;i++){
+            rByte.add(i);
+        }
+        return rByte;
+    }
+
+    public static Range<Short> of( short element1, short element2){
+        Range <Short> rShort = new Range<Short>();
+        for(short i=element1;i<=element2;i++){
```

Comment:
Think how you can improve code by calling of(T start, T end, Function<T, T> function)